src/Components/MainTableView/TableView.tsx

```diff
@@ -28,89 +39,139 @@ class TableView extends React.Component<{token: string, selectedSchemaName: stri
-            this.setState({tableAttributesInfo: undefined, errorMessage: 'Problem fetching table attributes: ' + error, isLoading: false})
-          })
+    // Check if table change, if so fetch the new data
+    if (prevProps.selectedTableName !== this.props.selectedTableName) {
+
+      // Determine what view the user is on and fetch accordingly
+      if (this.state.currentView === CurrentView.TABLE_CONTENT) {
+        // User is on TableContent, fetch data related to that view and set tableInfoNeedRefresh to true
+        this.fetchTableAttributeAndContent();
+        this.setState({tableContentNeedRefresh: false, tableDefinitionNeedRefresh: true});
       }
-      if (this.state.currentView === 'tableInfo') {
-        fetch(`${process.env.REACT_APP_DJLABBOOK_BACKEND_PREFIX}/get_table_definition`, {
-          method: 'POST',
-          headers: { 'Content-Type': 'application/json', 'Authorization': 'Bearer ' + this.props.token },
-          body: JSON.stringify({ schemaName: this.props.selectedSchemaName, tableName: this.props.selectedTableName })
-        })
-          .then(result => {
-            if (!result.ok) {
-              result.text()
-              .then(errorMessage => {
-                throw Error(`${result.status} - ${result.statusText}: (${errorMessage})`)
-              })
-              .catch(error => {
-                this.setState({tableInfoData: '', errorMessage: 'Problem fetching table information: ' + error})
-              })
-            }
-            return result.text()})
-          .then(result => {
-            this.setState({tableInfoData: result, errorMessage: ''})
-          })
-          .catch(error => {
-            this.setState({tableInfoData: '', errorMessage: 'Problem fetching table information: ' + error})
-          })
+      else if (this.state.currentView === CurrentView.TABLE_INFO) {
+        // User is on TableInfo, fetch data related to that view nad set tableContentNeedRefresh to true
+        this.fetchTableDefinition();
+        this.setState({tableContentNeedRefresh: true, tableDefinitionNeedRefresh: false});
+      }
+    }
+    else if (this.state.currentView !== prevState.currentView) {
+      // The view change thus update accordingly if refresh is needed
+      if (this.state.currentView === CurrentView.TABLE_CONTENT && this.state.tableContentNeedRefresh) {
+        // Fetch data realted to TableContent
+        this.fetchTableAttributeAndContent();
+        this.setState({tableContentNeedRefresh: false})
+      }
+      else if (this.state.currentView === CurrentView.TABLE_INFO && this.state.tableDefinitionNeedRefresh) {
+        // Fetch data related to TableInfo
+        this.fetchTableDefinition();
+        this.setState({tableDefinitionNeedRefresh: false})
       }
     }
+    else if (this.state.currentPageNumber !== prevState.currentPageNumber) {
+      this.fetchTableContent();
+    }
+    else if (this.state.numberOfTuplesPerPage != prevState.numberOfTuplesPerPage) {
```

Comment:
We should also implement here a ~1[s] delay before updating; similar to how it is done in filters. For example, if a user was updating it to `150` and took a slight pause between each digit.

Reply:
done